src/code/client.coffee

```diff
@@ -362,11 +365,14 @@ class CloudFileManagerClient
     if @_autoSaveInterval
       clearInterval @_autoSaveInterval
 
+    shouldAutoSave = =>
+      @state.dirty and not @state.metadata?.autoSaveDisabled and not @isSaveInProgress() and @state.metadata?.provider?.can 'save'
+
```

Comment:
Just a very minor nit: I think if you hoisted shouldAutoSave up into the class definition the setInterval() below would save a tiny amount of memory as the runtime wouldn't need to maintain the closure around autosave().